test/Model.js

```diff
@@ -3748,6 +3748,238 @@ describe("Model", () => {
+			}, "output": [{"id": 1, "name": "Bob"}, {"id": 2, "name": "Tim"}]},
+			{"input": () => {
+				User.serializer.add("isActive", {
+					"modify": (serialized, original) => {
+						serialized.isActive = original.status === "active";
+						return serialized;
+					}
+				});
+				return [[{"id": 1, "status": "active", "name": "Bob"}, {"id": 2, "status": "not_active", "name": "Tim"}], "isActive"];
+			}, "output": [{"id": 1, "status": "active", "isActive": true, "name": "Bob"}, {"id": 2, "status": "not_active", "isActive": false, "name": "Tim"}]},
+			{"input": () => {
+				User.serializer.add("isActive", {
+					"exclude": ["status"],
+					"modify": (serialized, original) => {
+						serialized.isActive = original.status === "active";
+						return serialized;
+					}
+				});
+				return [[{"id": 1, "status": "active", "name": "Bob"}, {"id": 2, "status": "not_active", "name": "Tim"}], "isActive"];
+			}, "output": [{"id": 1, "isActive": true, "name": "Bob"}, {"id": 2, "isActive": false, "name": "Tim"}]},
+			{"input": () => {
+				User.serializer.add("isActive", {
+					"include": ["id"],
+					"modify": (serialized, original) => {
+						serialized.isActive = original.status === "active";
+						return serialized;
+					}
+				});
+				return [[{"id": 1, "status": "active", "name": "Bob"}, {"id": 2, "status": "not_active", "name": "Tim"}], "isActive"];
+			}, "output": [{"id": 1, "isActive": true}, {"id": 2, "isActive": false}]},
+			{"input": [[{"id": 1, "name": "Bob"}, {"id": 2, "name": "Tim"}], "random"], "error": "Field options is required and should be an object or array"},
+			{"input": [{"id": 1, "name": "Bob"}], "error": "documentsArray must be an array of document objects"},
+		];
+		describe("Model.serializeMany", () => {
+			it("Should be a function", () => {
+				expect(User.serializeMany).to.be.a("function");
+			});
+
+			serializeTests.forEach((test) => {
+				it(`Should return ${JSON.stringify(test.output)} for ${JSON.stringify(test.input)}`, () => {
+					const input = typeof test.input === "function" ? test.input() : test.input;
+					if (test.error) {
+						expect(() => User.serializeMany(...input)).to.throw(test.error);
+					} else {
+						expect(User.serializeMany(...input)).to.eql(test.output);
+					}
+				});
+
+				it(`Should return ${JSON.stringify(test.output)} for ${JSON.stringify(test.input)} when using document instance`, () => {
+					const input = typeof test.input === "function" ? test.input() : test.input;
+					if (Array.isArray(input[0])) {
+						input[0] = input[0].map((obj) => new User(obj));
+					}
+
+					if (test.error) {
+						expect(() => User.serializeMany(...input)).to.throw(test.error);
+					} else {
+						expect(User.serializeMany(...input)).to.eql(test.output);
+					}
+				});
+			});
+		});
+
+		describe("model.serialize", () => {
+			it("Should be a function", () => {
+				expect(new User().serialize).to.be.a("function");
+			});
+
+			serializeTests.forEach((test) => {
```

Comment:
I really wish this could be part of `test/Document.js`, but I can't figure out a clean way to make it DRY while still doing that.

Any ideas anyone?